kubectl/Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM alpine:3.10
+LABEL maintainer="Maxime Sibellas <maxime.sibellas@ekino.com>"
+
+ARG KUBECTL_VERSION
+ARG KUSTOMIZATION_VERSION
+ARG KUBECTX_VERSION
+ARG KUBESCORE_VERSION
+
+ENV PATH="${PATH}:/usr/bin"
+
+RUN apk add ca-certificates curl bash wget ncurses && \
+    curl -LO https://storage.googleapis.com/kubernetes-release/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl && \
+    chmod +x ./kubectl && \
+    mv ./kubectl /usr/bin/kubectl
+
+RUN wget https://github.com/kubernetes-sigs/kustomize/releases/download/v${KUSTOMIZATION_VERSION}/kustomize_${KUSTOMIZATION_VERSION}_linux_amd64 && \
```

Comment:
Better to have only one `RUN` instruction

Reply:
Why ? This way we can use docker layer sharing properties. As images are downloaded on each gitlab runners, a new version will not require a full download of a new image, only some layers.

Reply:
[You should minimize your number of layers.
](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers)

It's an old school rule of thumb, and every official image used to be required to follow it too, I don't if it's still the case.

My understanding is that while multiple layers are cool for cacheability (with a caveat,  see below), you produce heavier images and you can't control you image size as efficiently (you can't remove in layer #2 something that you downloaded in layer #1).

And the cache is  not that cool anyway : if you have ten layers and your layer #2 changes, you'll have to rebuild #3 to #10 again.

The new multiple `FROM` is hella cool though :sunglasses:

Reply:
What I understand of that rule is : if you have to compile a tool from sources (download + compile + delete sources), you will have to define all steps inside one `RUN` statement to avoid increasing the weight of the image.

But when you just download binaries, you may want to user docker layers chacheability. And to use it efficiently, you have to order your layer from the least prone to change to the one who will change more often. 

Bonus: multiple layers are way easier to debug when there is a lot of dependencies to install.

Reply:
Still, it makes no sense to have 2 layers for `kubectx` and `kubens` since they use the same version.

Reply:
Yes, I merged kubectx and kubens RUN statement